report/src/main/kotlin/com/hartwig/actin/report/pdf/chapters/EfficacyEvidenceChapter.kt

```diff
@@ -1,28 +1,106 @@
 package com.hartwig.actin.report.pdf.chapters
 
+import com.hartwig.actin.configuration.EfficacyEvidenceChapterType
+import com.hartwig.actin.configuration.ReportConfiguration
+import com.hartwig.actin.datamodel.algo.AnnotatedTreatmentMatch
+import com.hartwig.actin.datamodel.algo.ShapDetail
+import com.hartwig.actin.datamodel.algo.SimilarPatientsSummary
+import com.hartwig.actin.datamodel.personalization.MeasurementType
 import com.hartwig.actin.report.datamodel.Report
+import com.hartwig.actin.report.pdf.tables.TableGeneratorFunctions
+import com.hartwig.actin.report.pdf.tables.molecular.MolecularEfficacyDescriptionGenerator
+import com.hartwig.actin.report.pdf.tables.molecular.OffLabelMolecularClinicalEvidenceGenerator
+import com.hartwig.actin.report.pdf.tables.molecular.OnLabelMolecularClinicalEvidenceGenerator
+import com.hartwig.actin.report.pdf.tables.molecular.TreatmentRankingGenerator
+import com.hartwig.actin.report.pdf.tables.soc.EfficacyEvidenceDetailsGenerator
 import com.hartwig.actin.report.pdf.tables.soc.EfficacyEvidenceGenerator
+import com.hartwig.actin.report.pdf.tables.soc.RealWorldSurvivalOutcomesGenerator
+import com.hartwig.actin.report.pdf.tables.soc.RealWorldTreatmentDecisionsGenerator
+import com.hartwig.actin.report.pdf.tables.soc.ResistanceEvidenceGenerator
 import com.hartwig.actin.report.pdf.util.Cells
+import com.hartwig.actin.report.pdf.util.Styles
 import com.hartwig.actin.report.pdf.util.Tables
+import com.hartwig.actin.treatment.EvidenceScoringModel
+import com.hartwig.actin.treatment.TreatmentRankingModel
+import com.hartwig.actin.treatment.createScoringConfig
 import com.itextpdf.kernel.geom.PageSize
 import com.itextpdf.layout.Document
+import com.itextpdf.layout.element.Image
+import com.itextpdf.layout.element.Paragraph
+import com.itextpdf.layout.element.Table
+import com.itextpdf.svg.converter.SvgConverter
+import org.apache.logging.log4j.LogManager
+import org.jetbrains.letsPlot.Stat
+import org.jetbrains.letsPlot.export.ggsave
+import org.jetbrains.letsPlot.geom.geomBar
+import org.jetbrains.letsPlot.geom.geomLine
+import org.jetbrains.letsPlot.ggsize
+import org.jetbrains.letsPlot.label.ggtitle
+import org.jetbrains.letsPlot.label.labs
+import org.jetbrains.letsPlot.letsPlot
+import org.jetbrains.letsPlot.pos.positionDodge
+import org.jetbrains.letsPlot.scale.guides
+import org.jetbrains.letsPlot.scale.scaleFillManual
+import org.jetbrains.letsPlot.scale.scaleYContinuous
+import java.io.ByteArrayInputStream
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.createTempFile
+import kotlin.io.path.readBytes
+import kotlin.math.abs
+import kotlin.math.sign
 
-class EfficacyEvidenceChapter(private val report: Report, override val include: Boolean) : ReportChapter {
+class EfficacyEvidenceChapter(private val report: Report, private val configuration: ReportConfiguration) : ReportChapter {
```

Comment:
I see the advantage logically to consolidate the chapters, but the result is a quite a complex chapter that is hard to test and read. I would consider trying to decompose this into more modular pieces (although they can stay part of the same chapter)